thumbor/context.py

```diff
@@ -69,10 +69,17 @@ def __enter__(self):
         return self
 
     def __exit__(self, exception_type, value, traceback):
-        if self.modules:
+        if self.thread_pool:
+            self.thread_pool.cleanup()
+
+        self.cleanup()
+
+    def cleanup(self):
+        if hasattr(self, "modules") and self.modules:
```

Comment:
This line — and probably line 81 as well, more or less in the same way — could read like the following:

```suggestion
        if getattr(self, "modules", None):
```